app/models/host/base.rb

```diff
@@ -175,7 +173,8 @@ def set_interfaces(parser)
       end
 
       parser.interfaces.each do |name, attributes|
-        iface = get_interface_scope(name, attributes).first || interface_class(name).new(:managed => false)
+        iface_scope = get_interface_scope(name, attributes)
```

Comment:
The link doesn't show anything for me, and I don't have that commit SHA. All code paths from `get_interface_scope` should return a scope, so please expand on why that returns nil? This appears to be working around a bug elsewhere.

As already noted, the unit test would assist.